src/fsipy/simulations/simulation_common.py

```diff
@@ -0,0 +1,274 @@
+    Q_mean = info['mean_flow_rate']
+    area_ratio = info['area_ratio']
+    area_inlet = info['inlet_area']
+
+    return id_in, id_out, id_wall, Q_mean, area_ratio, area_inlet
+
+
+def load_probe_points(mesh_path: Union[str, Path]) -> np.ndarray:
+    """
+    Load probe points from a corresponding file based on the mesh file's path.
+
+    Args:
+        mesh_path (str or Path): The path to the mesh file.
+
+    Returns:
+        np.ndarray: An array containing the loaded probe points.
+    """
+    mesh_path = Path(mesh_path)
+    rel_path = mesh_path.stem + "_probe_point"
+    probe_points_path = mesh_path.parent / rel_path
+    probe_points = np.load(probe_points_path, encoding='latin1', fix_imports=True, allow_pickle=True)
+
+    return probe_points
+
+
+def print_probe_points(v: Function, p: Function, probe_points: List[Union[float, np.ndarray]]) -> None:
+    """
+    Print velocity and pressure at probe points.
+
+    Args:
+        v (dolfin.Function): Velocity function with components v.sub(0), v.sub(1), and v.sub(2).
+        p (dolfin.Function): Pressure function.
+        probe_points (list): List of probe points.
+
+    Returns:
+        None
+    """
+    for i, point in enumerate(probe_points):
+        # Extract components of velocity and pressure at the probe point
+        uu = peval(v.sub(0), point)
+        vv = peval(v.sub(1), point)
+        ww = peval(v.sub(2), point)
+        pp = peval(p, point)
+
+        if MPI.rank(MPI.comm_world) == 0:
+            print(f"Probe Point {i}: Velocity: ({uu}, {vv}, {ww}) | Pressure: {pp}")
+
+
+def peval(f: Function, x: Union[float, np.ndarray]) -> np.ndarray:
+    """
+    Parallel synchronized evaluation of a function.
+
+    Args:
+        f (dolfin.Function): Function to be evaluated.
+        x (Union[float, np.ndarray]): Input value for the function.
+
+    Returns:
+        np.ndarray: Evaluated function values.
+    """
+    try:
+        yloc = f(x)
+    except RuntimeError:
+        yloc = np.inf * np.ones(f.value_shape())
+
+    comm = MPI.comm_world
+    yglob = np.zeros_like(yloc)
+    comm.Allreduce(yloc, yglob, op=mpi.MIN)
+
+    return yglob
+
+
+def calculate_and_print_flow_properties(dt: float, mesh: Mesh, v: Function, inlet_area: float, mu_f: float,
+                                        n: Expression, dsi: Measure) -> None:
+    """
+    Calculate and print flow properties.
+
+    Args:
+        dt (float): Time step size.
+        mesh (dolfin.Mesh): Mesh object.
+        v (dolfin.Function): Velocity field.
+        inlet_area (float): Inlet area.
+        mu_f (float): Fluid dynamic viscosity.
+        n (dolfin.Expression): FacetNormal expression.
+        dsi (dolfin.Measure): Measure for inlet boundary.
+
+    Returns:
+        None
+    """
+    # Calculate the DG vector of velocity magnitudes
+    DG = FunctionSpace(mesh, "DG", 0)
+    V_vector = project(sqrt(inner(v, v)), DG).vector().get_local()
+    h = mesh.hmin()
```

Comment:
This `calculate_and_print_flow_properties` is definitely useful, but I think we should avoid using `project` inside the function that might be used many times during the time loop. This is because `project` is a very costly function   that can be easily optimized. For reference, @jorgensd ’s [tutorial](https://simula-sscp.github.io/SSCP_2023_lectures/L12%20%28FEniCS%20Intro%29/L03_FEniCS_Darcy.html#advanced-reusable-projection) shows how to make reusable projection.

Reply:
Especially if we  are projecting into `DG`-0 we should use the local projection, as no global inversion of any matrices are needed.

Reply:
@keiyamamo and @jorgensd - I have added Jørgen's `local_project` function and use this instead of `project` in `calculate_and_print_flow_properties`. Let me know if this is the correct way to do it.

Reply:
Looks good to me.

Reply:
@jorgensd  Would it be possible to combine reusable and local projection? or Doesn’t it really make difference in case of local projection ?  I can see that `local_project` function still creates `a_proj` every time even though it is constant.

Reply:
No, i couldnt find a way of doing the Pre-assembly of the lhs with localsolver. This is a limitation of Dolfin. 

One could have an option to either use a global matrix (the reusable projector, that works with any function space), or the local projector (that only works on DG elements.

Reply:
Yes, But it would require some testing, as I not 100% confident local solve is always the right thing to use. We might be better of using a re-usable (global) projection class if we want things to be guaranteed to do what we expect.

Reply:
I see. I just want to make sure that projection is as cheap as possible since this function `calculate_and_print_flow_properties` is supposed to be called many times during the simulation. I might compute the time difference between local and re-usable(global) projection and see which one is faster. For now, I think we can keep what Johannes made :)